Tests/Runtime/CmColliderTests.cs

```diff
@@ -44,56 +52,73 @@ public IEnumerator CheckSmoothingTime()
             yield return null;
             Assert.That(originalCamPosition == m_Vcam.State.FinalPosition, Is.True);
 
-            var obstacle = GameObject.CreatePrimitive(PrimitiveType.Cube);
-            obstacle.transform.position = originalCamPosition; // put obstacle in the way
+            var obstacle = CreateObstacle(originalCamPosition);
 
+            yield return null;
+            yield return null;
             yield return null;
             Assert.That(originalCamPosition == m_Vcam.State.FinalPosition, Is.False);
             
             obstacle.transform.position = originalCamPosition + Vector3.left * 100f; // move obstacle out of the way
 
+            yield return null;
+            yield return null;
+            yield return null;
             yield return new WaitForSeconds(1);
             Assert.That(originalCamPosition == m_Vcam.State.FinalPosition, Is.True);
         }
         
         [UnityTest]
         public IEnumerator CheckDampingWhenOccluded()
         {
-            m_Collider.m_DampingWhenOccluded = 10;
+            m_Collider.m_DampingWhenOccluded = 5;
             var originalCamPosition = m_Vcam.State.FinalPosition;
 
             yield return null;
             Assert.That(originalCamPosition == m_Vcam.State.FinalPosition, Is.True);
 
-            var obstacle = GameObject.CreatePrimitive(PrimitiveType.Cube);
-            obstacle.transform.position = originalCamPosition; // put obstacle in the way
+            var obstacle = CreateObstacle(originalCamPosition);
 
+            yield return null;
+            yield return null;
             yield return null;
             Assert.That(originalCamPosition == m_Vcam.State.FinalPosition, Is.False);
             var pos1 = m_Vcam.State.FinalPosition;
 
+            yield return null;
+            yield return null;
             yield return null;
             Assert.That(pos1 == m_Vcam.State.FinalPosition, Is.False);
+            
+            obstacle.transform.position = originalCamPosition + Vector3.left * 100f; // move obstacle out of the way
+
+            yield return null;
+            yield return null;
+            yield return null;
+            Assert.That(originalCamPosition == m_Vcam.State.FinalPosition, Is.True);
```

Comment:
Do you really want hard equality here? (and elsewhere in this file)

Reply:
Yes, I expect exact equality. Though, I could add some tolerance. I ran the tests hundreds of times, and it was ok with exact equality.